### PR TITLE
Make yamllint consistent with Composer

### PR DESCRIPTION
**Pull request by b-chu**
Matches the yaml settings for Composer
https://github.com/mosaicml/composer/blob/dev/.yamllint.yaml

### CODE REVIEW

**Comment by karan6181** [comment]
@b-chu Thanks, Brian, for creating these changes. We will hold this PR off for a week or so since there are huge changes coming into the `main` for the Delta Table reader and then we will re-base this on top of `main` after that.

**Review by snarayan21 — APPROVED** [review]
lgtm, pending tests passing!

**Pull request merged**
